src/sentry/integrations/gitlab/webhooks.py

```diff
@@ -193,75 +193,101 @@ def dispatch(self, request: Request, *args, **kwargs) -> Response:
-            return HttpResponse(
-                status=400,
-                reason="Gitlab sent us a malformed HTTP_X_GITLAB_TOKEN.",
-            )
+            return HttpResponse(status=400, reason=extra["reason"])
         except Exception as e:
+            logger.info("gitlab.webhook.invalid-token", extra=extra)
+            extra["reason"] = "Generic catch-all error."
             capture_exception(e)
-            logger.info("gitlab.webhook.invalid-token", extra={"token": token})
-            return HttpResponse(status=400, reason="Generic catch-all error.")
+            return HttpResponse(status=400, reason=extra["reason"])
 
         try:
             integration = (
-                Integration.objects.filter(provider=self.provider, external_id=external_id)
+                Integration.objects.filter(
+                    provider=self.provider,
+                    external_id=external_id,  # e.g. example.gitlab.com:group-x
+                )
                 .prefetch_related("organizations")
                 .get()
             )
-        except Integration.DoesNotExist:
-            logger.info(
-                "gitlab.webhook.invalid-organization",
-                extra={"external_id": request.META["HTTP_X_GITLAB_TOKEN"]},
-            )
-            return HttpResponse(status=400)
+            extra = {
+                **extra,
+                **{
+                    # I'm naming it like this to have both side by side in the breadcrumbs
+                    "tooken_from_sentry": f"{integration.external_id}:{integration.metadata['webhook_secret']}",
+                    "integration": {
+                        # The metadata could be useful to debug
+                        # domain_name -> gitlab.com/getsentry-ecosystem/foo'
+                        # scopes -> ['api']
+                        "metadata": integration.metadata,
+                        "id": integration.id,  # This is useful to query via Redash
+                        "status": integration.status,  # 0 seems to be active
+                    },
+                    # I do not know how we could have multiple integration installation to many organizations
+                    "slugs": ",".join(map(lambda x: x.slug, integration.organizations.all())),
+                },
+            }
+        except Integration.DoesNotExist as e:
+            logger.info("gitlab.webhook.invalid-organization", extra=extra)
+            extra["reason"] = "There is not integration that matches your organization."
+            capture_exception(e)
+            return HttpResponse(status=400, reason=extra["reason"])
 
-        if not constant_time_compare(secret, integration.metadata["webhook_secret"]):
-            logger.info(
-                "gitlab.webhook.invalid-token-secret", extra={"integration_id": integration.id}
-            )
-            # XXX: Look into using set_context
-            capture_message("Gitlab's webhook secret does not match.")
```

Comment:
Switching to `capture_error` in order to get a stack trace so we can get to the code.